### PR TITLE
Fix various minor issues in docstrings

### PR DESCRIPTION
**Pull request by dmontagu**

### CODE REVIEW

File: pydantic/fields.py [hunk 1]
```diff
@@ -240,36 +240,34 @@ class MyModel(pydantic.BaseModel):
...
-        Example:
-            This is how you can create a field from a bare annotation like this:
+        class MyModel(pydantic.BaseModel):
+            foo: int  # <-- like this
+        ```
 
-            ```python
-            import pydantic
+        We also account for the case where the annotation can be an instance of `Annotated` and where
+        one of the (not first) arguments in `Annotated` are an instance of `FieldInfo`, e.g.:
```
Comment (sydney-runkle) [inline pydantic/fields.py:253]:
```suggestion
        We also account for the case where the annotation can be an instance of `Annotated` and where
        one of the (not first) arguments in `Annotated` is an instance of `FieldInfo`, e.g.:
```

Comment (sydney-runkle) [reply pydantic/fields.py:253]: I think... (ex: one of the ducks _is_ in the pond).

File: pydantic/fields.py [hunk 2]
```diff
@@ -653,23 +653,25 @@ def Field(  # noqa: C901
...
         default: Default value if the field is not set.
         default_factory: A callable to generate the default value, such as :func:`~datetime.utcnow`.
-        alias: An alternative name for the attribute.
+        alias: The name to use for the attribute when validating or serializing by alias.
+            This is often used for things like converting between snake and camel case.
         alias_priority: Priority of the alias. This affects whether an alias generator is used.
-        validation_alias: 'Whitelist' validation step. The field will be the single one allowed by the alias or set of
-            aliases defined.
-        serialization_alias: 'Blacklist' validation step. The vanilla field will be the single one of the alias' or set
-            of aliases' fields and all the other fields will be ignored at serialization time.
```
Comment (sydney-runkle) [inline pydantic/fields.py:661]: Begone! 😎

File: pydantic/fields.py [hunk 3]
```diff
@@ -653,23 +653,25 @@ def Field(  # noqa: C901
...
         discriminator: Field name or Discriminator for discriminating the type in a tagged union.
         json_schema_extra: A dict or callable to provide extra JSON schema properties.
-        frozen: Whether the field is frozen.
-        validate_default: Run validation that isn't only checking existence of defaults. This can be set to `True` or `False`. If not set, it defaults to `None`.
+        frozen: Whether the field is frozen. If true, attempts to change the value on an instance will raise an error.
+        validate_default: If `True`, apply validation to the default value every time you create an instance.
+            Otherwise, for performance reasons, the default value of the field is trusted and not validated.
         repr: A boolean indicating whether to include the field in the `__repr__` output.
         init_var: Whether the field should be included in the constructor of the dataclass.
+            (Only applies to dataclasses.)
```
Comment (sydney-runkle) [inline pydantic/fields.py:672]: I assume this is the correct amount of indentation?

File: pydantic/fields.py [hunk 4]
```diff
@@ -679,20 +681,20 @@ def Field(  # noqa: C901
         multiple_of: Value must be a multiple of this. Only applicable to numbers.
         min_length: Minimum length for strings.
         max_length: Maximum length for strings.
-        pattern: Pattern for strings.
+        pattern: Pattern for strings (a regular expression).
```
Comment (sydney-runkle) [inline pydantic/fields.py:684]: Should we say anything here about configuring the `regex_engine`? Fine if not, bc I believe that's at the `model_config` level.

File: pydantic/fields.py [hunk 5]
```diff
@@ -894,13 +902,13 @@ class ComputedFieldInfo:
         decorator_repr: A class variable representing the decorator string, '@computed_field'.
         wrapped_property: The wrapped computed field property.
         return_type: The type of the computed field property's return value.
-        alias: The alias of the property to be used during encoding and decoding.
-        alias_priority: priority of the alias. This affects whether an alias generator is used
-        title: Title of the computed field as in OpenAPI document, should be a short summary.
-        description: Description of the computed field as in OpenAPI document.
-        examples: Example values of the computed field as in OpenAPI document.
+        alias: The alias of the property to be used during serialization.
+        alias_priority: The priority of the alias. This affects whether an alias generator is used.
```
Comment (sydney-runkle) [inline pydantic/fields.py:906]: Should we talk about different levels here? Or leave that to the usage docs?

File: pydantic/json_schema.py [hunk 6]
```diff
@@ -88,7 +88,7 @@
 
 
 def update_json_schema(schema: JsonSchemaValue, updates: dict[str, Any]) -> JsonSchemaValue:
-    """Update a JSON schema by providing a dictionary of updates.
+    """Update a JSON schema in-place by providing a dictionary of updates.
```
Comment (sydney-runkle) [inline pydantic/json_schema.py:91]: Just a side note - isn't it bad practice to both:
1. Modify an input argument in-place
2. Return said argument

Shouldn't we do one or the other? As in, either modify the `schema` in-place, then return `None`, or create a deepcopy of the `schema`, modify it, then return that copy?

Comment (sydney-runkle) [reply pydantic/json_schema.py:91]: Obviously not related to this docs PR, but just came to my attention.

File: pydantic/json_schema.py [hunk 7]
```diff
@@ -757,8 +750,9 @@ def literal_schema(self, schema: core_schema.LiteralSchema) -> JsonSchemaValue:
             return {'enum': expected}
 
     def is_instance_schema(self, schema: core_schema.IsInstanceSchema) -> JsonSchemaValue:
-        """Generates a JSON schema that checks if a value is an instance of a class, equivalent to Python's
-        `isinstance` method.
+        """Handles JSON schema generation for a core schema that checks if a value is an instance of a class.
+
+        Unless overridden in a subclass, this raises an error.
```
Comment (sydney-runkle) [inline pydantic/json_schema.py:755]: What kind of error / why?

Comment (sydney-runkle) [reply pydantic/json_schema.py:755]: A JSON schema generation error I presume. Ignore this.

File: pydantic/type_adapter.py [hunk 8]
```diff
@@ -113,7 +112,7 @@ class TypeAdapter(Generic[T]):
     A `TypeAdapter` instance exposes some of the functionality from `BaseModel` instance methods
     for types that do not have such methods (such as dataclasses, primitive types, and more).
 
-    Note that `TypeAdapter` is not an actual type, so you cannot use it in type annotations.
+    **Note:** `TypeAdapter` instances are not types, and cannot be used as type annotations for fields.
```
Comment (sydney-runkle) [inline pydantic/type_adapter.py:115]: Nice, I'll make sure to refer to this when the weekly type adapter issue / question pops up.

**Review by sydney-runkle — APPROVED** [review]
LGTM overall, just a few questions and one or two suggestions 👍.

Thanks for doing this!

File: pydantic/fields.py (hunk 5 above)
Comment (dmontagu) [reply pydantic/fields.py:906]: 🤷‍♂️ I think it's okay if we just leave it to usage docs, if someone wants to make a change they can, I certainly wouldn't object just didn't want to take a stand in this PR

File: pydantic/fields.py (hunk 4 above)
Comment (dmontagu) [reply pydantic/fields.py:684]: sounds good to me

File: pydantic/json_schema.py (hunk 6 above)
Comment (dmontagu) [reply pydantic/json_schema.py:91]: In general yes but I think this is somehow related to some backwards compatibility thing. I don't remember why I added a function for this all that time ago lol, but I would have looked at it and been like "what is the point of this" so I feel like there was one

Comment (dmontagu) [reply pydantic/json_schema.py:91]: Okay I dug into this a bit, and I think this function used to be used in more places (I added it over 8 months ago, well before pydantic v2 was even released), and now it isn't. I think we should open a PR to deprecate it (sadly it looks like it's technically public API.. 😩), and remove all internal usage of it.

Comment (sydney-runkle) [reply pydantic/json_schema.py:91]: Cool beans, I'll open an issue.

Comment (sydney-runkle) [reply pydantic/json_schema.py:91]: https://github.com/pydantic/pydantic/issues/8381

**Pull request merged**